node/src/actors/chain_manager/handlers.rs

```diff
@@ -284,11 +285,21 @@ impl Handler<AddBlocks> for ChainManager {
             "AddBlocks received while StateMachine is in state {:?}",
             self.sm_state
         );
+
+        let consensus_constants = self
+            .chain_state
+            .chain_info
+            .as_ref()
+            .unwrap()
+            .consensus_constants
+            .clone();
```

Comment:
You probably can avoid this clone by using a reference:

    let consensus_constants = &self